gtsfm/frontend/matcher/twoway_matcher.py

```diff
@@ -60,28 +74,54 @@ def match(
             distance_metric = cv.NORM_HAMMING
         else:
             raise NotImplementedError("The distance type is not in MatchingDistanceType")
-
         if descriptors_i1.size == 0 or descriptors_i2.size == 0:
             return np.array([])
 
-        # we will have to remove NaNs by ourselves
+        # Remove NaNs.
         valid_idx_i1 = np.nonzero(~(np.isnan(descriptors_i1).any(axis=1)))[0]
         valid_idx_i2 = np.nonzero(~(np.isnan(descriptors_i2).any(axis=1)))[0]
 
         descriptors_1 = descriptors_i1[valid_idx_i1]
         descriptors_2 = descriptors_i2[valid_idx_i2]
 
-        # run OpenCV's matcher
-        bf = cv.BFMatcher(normType=distance_metric, crossCheck=True)
-        matches = bf.match(descriptors_1, descriptors_2)
-        matches = sorted(matches, key=lambda r: r.distance)
-
+        # Run KNN matcher both ways.
+        bf_matcher = cv.BFMatcher(normType=distance_metric)
+        init_matches1 = bf_matcher.knnMatch(descriptors_1, descriptors_2, k=2)
+        init_matches2 = bf_matcher.knnMatch(descriptors_2, descriptors_1, k=2)
```

Comment:
Why not let opencv handle crosscheck?

Reply:
Because it does not compute the distance to the next nearest neighbor for the ratio test